DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -185,6 +192,32 @@ public void SetCompletionDate(int progressId, DateTime? completionDate)
             }
         }
 
+        public void UpdateDiagnosticScore(int progressId, int tutorialId, int myScore)
+        {
+            var numberOfAffectedRows = connection.Execute(
+                @"UPDATE ap
+                        SET ap.DiagHigh = IIF(@myScore > DiagHigh, @myScore, DiagHigh),
+                            ap.DiagLow = CASE WHEN ap.DiagAttempts = 0 THEN @myScore
+                                ELSE IIF(@myScore > DiagLow, DiagLow, @myScore) END,
+                            ap.DiagLast = @myScore,
+                            ap.DiagAttempts = DiagAttempts + 1
+                        FROM aspProgress AS ap
+                        INNER JOIN Tutorials AS t ON t.TutorialID = ap.TutorialID
+                        WHERE ap.ProgressID = @progressId
+                        AND (ap.TutorialID = @tutorialId OR t.OriginalTutorialID = @tutorialId)",
+                new { progressId, tutorialId, myScore }
+            );
+
+            if (numberOfAffectedRows < 1)
+            {
+                logger.LogWarning(
+                    "Not updating diagnostic score as db update failed. " +
+                    $"Progress id: {progressId}, tutorial Id: {tutorialId}, myScore: {myScore}"
+                );
+                throw new Exception();

```

Comment:
I missed this. I think it would be good to include a message in the exception. From a logging perspective it will be a bit clearer as to why we have a failure.